src/main/java/Deadline.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Represents the attributes of a Deadline object, which is a sub-class of Task.
+ */
+public class Deadline extends Task {
+
+    protected String date;
+
+    /**
+     * Creates a Deadline object with a description.
+     * The identity is also intialised as D, for all Deadline objects.
+     *
+     * @param description of the Deadline task.
+     */
+    public Deadline(String description, String date) {
+        super(description);
+        this.date = date;
+        this.identity = 'D';
+    }
+
+    /**
+     * Recovery of a Deadline object based on the format of a task in the Duke.txt file that we save our information into.
+     * Whether the task is done or not depends on the int value of the task when it is saved.
+     *
+     * @param intDone if task is done, the int value == 1. If the task is not done, int value == 0.
+     * @param description of the task.
+     */
+    public Deadline(int intDone, String description, String date) {
```

Comment:
Creative way of handling the isDone attribute. Another alternative may be using a setIsDone method